modules/openapi-generator/src/main/resources/dart-dio/api.mustache

```diff
@@ -52,26 +52,15 @@ class {{classname}} {
         ];{{/hasConsumes}}
         {{#hasFormParams}}
 
-        final formData = <String, dynamic>{};
+        final formData = <String, dynamic>{
+            {{#formParams}}
+            {{^required}}if ({{paramName}} != null) {{/required}}r'{{baseName}}': {{#isFile}}MultipartFile.fromBytes({{paramName}}, filename: r'{{baseName}}'){{/isFile}}{{^isFile}}parameterToString(_serializers, {{paramName}}){{/isFile}},
```

Comment:
What about required fields that are nullable (`nullable: true`)? Do we need to handle it above?

Reply:
Well these are formParams and not fields, I have never seen them `nullable` but since only required fields can be `nullable: true` they would currently be empty strings `''` if they are `null`.

Reply:
optional fields can also have `nullable: true`, which means the field will explicitly emit `null`. For example in JSON:

```json
{
  "name": "something",
  "type": null
}
```
whereas `type` is defined as both optional and nullable.

Reply:
Hmm ok, but if `type` is optional then how will it ever be not present in the output if it is `nullable`? If `type` is set to `null` it needs to be serialized in JSON, if it is anything else it also needs to be serialized in JSON - that basically means it is not optional?

In any case I didn't see anything about parameter objects being `nullable` in the spec, although `CodegenParameter` has a `nullable` flag, so I am not sure if this needs to be handled here.

Reply:
Updated with an additional `{{^nullable}}` check.